aspnetcore/mvc/models/file-uploads/samples/2.x/SampleApp/Utilities/FileHelpers.cs

```diff
@@ -0,0 +1,261 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Net.Http.Headers;
+
+namespace SampleApp.Utilities
+{
+    public static class FileHelpers
+    {
+        // If you require a check on specific characters in the IsValidFileExtensionAndSignature
+        // method, supply the characters in the _allowedChars field.
+        private static readonly byte[] _allowedChars = { };
+        // For more file signatures, see the File Signatures Database: https://www.filesignatures.net/
+        private static readonly Dictionary<string, byte[]> _fileSignature = new Dictionary<string, byte[]>
+        {
+            { ".doc", new byte[] { 0xD0, 0xCF, 0x11, 0xE0, 0xA1, 0xB1, 0x1A, 0xE1 } },
+            { ".docx", new byte[] { 0x50, 0x4B, 0x03, 0x04 } },
+            { ".gif", new byte[] { 0x47, 0x49, 0x46, 0x38 } },
+            { ".jpeg", new byte[] { 0xFF, 0xD8, 0xFF, 0xE0, 0xFF, 0xD8, 0xFF, 0xE2, 0xFF, 0xD8, 0xFF, 0xE3 } },
+            { ".jpg", new byte[] { 0xFF, 0xD8, 0xFF, 0xE0, 0xFF, 0xD8, 0xFF, 0xE1, 0xFF, 0xD8, 0xFF, 0xE8 } },
+            { ".png", new byte[] { 0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A } },
+            { ".pdf", new byte[] { 0x25, 0x50, 0x44, 0x46 } },
+            { ".ppt", new byte[] { 0xD0, 0xCF, 0x11, 0xE0, 0xA1, 0xB1, 0x1A, 0xE1 } },
+            { ".pptx", new byte[] { 0x50, 0x4B, 0x03, 0x04 } },
+            { ".xls", new byte[] { 0xD0, 0xCF, 0x11, 0xE0, 0xA1, 0xB1, 0x1A, 0xE1 } },
+            { ".xlsx", new byte[] { 0x50, 0x4B, 0x03, 0x04 } },
+            { ".zip", new byte[] { 0x50, 0x4B, 0x03, 0x04, 0x50, 0x4B, 0x4C, 0x49, 0x54, 0x55, 0x50, 0x4B, 0x53, 0x70, 0x58, 0x50, 0x4B, 0x05, 0x06, 0x50, 0x4B, 0x07, 0x08, 0x57, 0x69, 0x6E, 0x5A, 0x69, 0x70 } },
+        };
+
+        // **WARNING!**
+        // In the following file processing methods, the file's content isn't scanned.
+        // In most production scenarios, an anti-virus/anti-malware scanner API is
+        // used on the file before making the file available to users or other
+        // systems. For more information, see the topic that accompanies this sample
+        // app.
+
+        public static async Task<byte[]> ProcessFormFile<T>(IFormFile formFile, 
+            ModelStateDictionary modelState, string[] permittedExtensions, 
+            long sizeLimit)
+        {
+            var fieldDisplayName = string.Empty;
+
+            // Use reflection to obtain the display name for the model
+            // property associated with this IFormFile. If a display
+            // name isn't found, error messages simply won't show
+            // a display name.
+            MemberInfo property =
```

Comment:
😕

Reply:
Can the user pass this in?

Reply:
AFAIK, this is the only way to get the display name for the model property ✨ *auto-magically* ✨. Do you think that reflection is really that bad to use/show? It would simplify this quite a bit to just have them pass it in. However, it would duplicate what they have for the model property's display name. This ended up in here because I wondered if I could just have my display names appear in **one place**. When I change it there, it changes it everywhere for every purpose. I don't mind the reflection; but if you prefer dropping it, then that's cool. I like *simple/straightforward stuff*, too.